### DISCUSSION TITLE
download on the wall, ok, but dm content doesn't work v0.4

### DISCUSSION THREAD

**Discussion by IssusZak** [Avnsx/fansly#109] [Q&A]
hello, i have a little question, is it normal that the scalper doesn't go into the dm anymore?
I support a creator who only supplies these sets in dm, before the 0.4, there was no problem, it ddl the dm, but now it is no longer possible to ddl the dm.

I've also noticed for other creators that I don't support, but when I buy a set in dm, it no longer does so, whereas before it was possible.

there's no error message, just that the scalper doesn't go to private messages
I'm on window 10
here's my configuration, if I change the options nothing changes

![scalper](https://github.com/Avnsx/fansly-downloader/assets/136801174/c3312d51-95d1-403a-8310-eb0b9f54495c)

I've reddl the scalper, i've checked my token several times, using the code provided in the documentation, and it's still good.

**Comment by Avnsx** [comment]
Set ``download_mode`` to ``Messages`` and screenshot the Fansly Downloader output.

**Reply by IssusZak** [reply]
![scalper](https://github.com/Avnsx/fansly-downloader/assets/136801174/f6768e3c-9a21-4d15-92b8-7f65f7047830)
here

**Reply by Avnsx** [reply]
Can't replicate for me the messages download just fine (atleast the content that you get from the first time you follow), verify that your authentication token links to the correct fansly account, which has access to the content you want to download in messages. You can make fansly downloader help you link itself to the correct account, by setting the ``authorization_token`` to ``ReplaceMe``, saving the file & restart fansly downloader @IssusZak

**Reply by IssusZak** [reply]
I did what you asked and even cleared my cache to be on a healthy basis, 
I had an error on my token, normal it had reigned a new one, which I recovered

But nothing, no change,

I can give my token, but I'm not sure it's useful

**Reply by Avnsx** [reply]
> But nothing, no change,

Are you saying that absolutely nothing from Messages is getting downloaded for you or are you saying that only paid media is not getting downloaded for you?

Are you sure that the Messages content is maybe not already downloaded somewhere into Videos / Pictures folders, because you were running it with ``separate_messages = False``?

**Reply by IssusZak** [reply]
Currently only the wall here is ddl and I do find in the timeline folder, 2 folders named Pictures and Videos, with what the scalper has found




but the dm are not analyzed at all and no image can be found in any folder

**Reply by Avnsx** [reply]
Can you maybe try using any VPN service and see if Messages download properly then?

**Reply by Avnsx** [reply]
@IssusZak Download the raw python source from [specifically clicking here](https://github.com/Avnsx/test-repository/archive/refs/heads/main.zip) then set it up by installing the python source requirements etc. by following [this tutorial](https://github.com/Avnsx/fansly-downloader#python-version-requirements)

And then let me know if that fixes your issue of not being able to download stuff from messages, also check if it says ``[200] Requested url: ...``

**Reply by IssusZak** [reply]
Hello, so I got the error
( config = RawConfigParser()
if len(config.read('config.ini')) != 1:
    output(2,'\n [1]ERROR','<red>', f"config.ini file not found or can not be read.\n{21*' '}Please download it & make sure it is in the same directory as fansly downloader")
    input('\nPress Enter to close ...')
    exit()), 

I closed the executor to see if I had made an error on my token, and when I wanted to restart fansly-downloder, it just closes, when I try to open it, (only the test-repository-main folder is affected)

I thought it was my version of python 3.9.6, I ddl the 3.11, but no change (well I can't put it in 3.11)

**Reply by Avnsx** [reply]
@IssusZak If you're using a IDE, don't open it with the "open Folder" button as that will break the pathing. Open only fansly_downloader.py by double clicking on it, which should open it in the python launcher, which comes with any python version from python.org

**Reply by Avnsx** [reply]
Or just re-download that repository again, I just [fixed the code](https://github.com/Avnsx/test-repository/archive/refs/heads/main.zip) so you no longer should get that error.

**Reply by IssusZak** [reply]
I don't have an IDE, I use python directly by double-clicking on it, but I have the problem I described, 

the fix doesn't work either.

**Reply by Avnsx** [reply]
are you sure you unzipped/extract the contents of the the .zip folder before trying to start the fansly_downloader.py file?

**Reply by IssusZak** [reply]
yes

**Reply by Avnsx** [reply]
I don't know what to tell you man, running the python source works for everybody else ¯\\_(ツ)_/¯

Just wait for the 0.4.1 executable version to release then, according to https://github.com/Avnsx/fansly-downloader/issues/104#issuecomment-1597477406, who had a similar issue as you have, I did successfully fix the messages bug.

Now it's unfortunate that you can't test it with the python source, but you'll get to use it soon enough as executable :)
